### PR TITLE
[Unitary hack] Improve Docstrings task/quera.py

### PR DESCRIPTION
**Pull request by shubhusion**
Fixes #957 
Improve Docstrings, Remove Markdown Errors

### CODE REVIEW

**Comment by shubhusion** [comment]
@Roger-luo please review this

File: src/bloqade/task/quera.py [hunk 1]
```diff
@@ -73,7 +115,15 @@ def fetch(self) -> "QuEraTask":
         return self
 
     def pull(self) -> "QuEraTask":
-        # blocking, force pulling, even its completed
+        """
+        Forcefully pulls the task results from the backend.
```
Comment (weinbe58) [inline src/bloqade/task/quera.py:119]:
```suggestion
        Blocks execution until task is in a stopped state, e.g. Completed, Cancelled, Failed, etc.
```

Wrong behavior, this function is used to Poll the remote API until the task is completed.

File: src/bloqade/task/quera.py [hunk 2]
```diff
@@ -131,12 +215,18 @@ def _result_exists(self) -> bool:
...
 def _serialze(obj: QuEraTask) -> Dict[str, ParamType]:
+    """
+    Serializes the QuEraTask object.
+
+    Args:
+        obj (QuEraTask): The QuEraTask object to serialize.
+
+    Returns:
+        Dict[str, ParamType]: The serialized QuEraTask object.
+    """
```
Comment (weinbe58) [inline src/bloqade/task/quera.py:229]: you don't need to document these private functions.

File: src/bloqade/task/quera.py [hunk 3]
```diff
@@ -155,6 +245,15 @@ def _serialze(obj: QuEraTask) -> Dict[str, ParamType]:
...
 def _deserializer(d: Dict[str, Any]) -> QuEraTask:
+    """
+    Deserializes a dictionary into a QuEraTask object.
+
+    Args:
+        d (Dict[str, Any]): The dictionary to deserialize.
+
+    Returns:
+        QuEraTask: The deserialized QuEraTask object.
+    """
```
Comment (weinbe58) [inline src/bloqade/task/quera.py:256]: same here.

**Review by weinbe58 — CHANGES_REQUESTED** [review]
Overall looks good, just need to address the comments.

**Comment by shubhusion** [comment]
@weinbe58 I have addressed the changes as requested by you

**Review by Roger-luo — CHANGES_REQUESTED** [review]
I think like the other PR, we should have some examples demonstrating how to use these APIs instead of just plain explanations.

**Comment by Roger-luo** [comment]
I'm going to comment on your other PRs with the same comment as following to avoid duplicated review comments for other reviewers:

I think you should work on #975 first instead of asking people to review other PRs, so we don't resolve duplicated issues in multiple PRs.

**Comment by shubhusion** [comment]
> I'm going to comment on your other PRs with the same comment as following to avoid duplicated review comments for other reviewers:
> 
> I think you should work on #975 first instead of asking people to review other PRs, so we don't resolve duplicated issues in multiple PRs.

@Roger-luo I just addressed all the requirements @weinbe58 mentioned. I am working on other PRs where you have mentioned changes related to examples.

**Pull request closed without merging**